From 8cbb5fe3833f0316f8cc45f718b1eee74caee6bd Mon Sep 17 00:00:00 2001
From: Dan Aloni <alonid@gmail.com>
Date: Fri, 24 Feb 2017 08:04:27 +0200
Subject: [PATCH] RPATH

---
 cmake/modules/AddLLDB.cmake | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/cmake/modules/AddLLDB.cmake b/cmake/modules/AddLLDB.cmake
index af16050769ad..103af9fbdf9b 100644
--- a/cmake/modules/AddLLDB.cmake
+++ b/cmake/modules/AddLLDB.cmake
@@ -92,8 +92,23 @@ macro(add_lldb_library name)
   set_target_properties(${name} PROPERTIES FOLDER "lldb libraries")
 endmacro(add_lldb_library)
 
+set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
+if (APPLE)
+  set(CMAKE_INSTALL_NAME_DIR "@rpath")
+  set(CMAKE_INSTALL_RPATH "@executable_path/../lib")
+else(UNIX)
+  if(NOT DEFINED CMAKE_INSTALL_RPATH)
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib${LLVM_LIBDIR_SUFFIX}")
+    if(${CMAKE_SYSTEM_NAME} MATCHES "(FreeBSD|DragonFly)")
+      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,origin")
+      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,origin")
+    endif()
+  endif(NOT DEFINED CMAKE_INSTALL_RPATH)
+endif()
+
 macro(add_lldb_executable name)
   add_llvm_executable(${name} DISABLE_LLVM_LINK_LLVM_DYLIB ${ARGN})
+  target_link_libraries(${name} ${CMAKE_EXECUTABLE_RPATH_LINK_CXX_FLAG}../../lib${LLVM_LIBDIR_SUFFIX} )
   set_target_properties(${name} PROPERTIES FOLDER "lldb executables")
 endmacro(add_lldb_executable)
 
-- 
2.7.4

