diff -up llvm-3.2.src/tools/llvm-shlib/Makefile.jx llvm-3.2.src/tools/llvm-shlib/Makefile
--- llvm-3.2.src/tools/llvm-shlib/Makefile.jx	2012-08-06 16:52:18.000000000 -0400
+++ llvm-3.2.src/tools/llvm-shlib/Makefile	2013-05-01 16:55:20.286980046 -0400
@@ -76,7 +76,7 @@ endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU))
     # Don't allow unresolved symbols.
-    LLVMLibsOptions += -Wl,--no-undefined
+    LLVMLibsOptions += -Wl,--no-undefined -Wl,-Bsymbolic
 endif
 
 ifeq ($(HOST_OS),SunOS)
