--- llvm-2.5/lib/System/Unix/Signals.inc.gcc44	2008-12-05 18:39:24.000000000 -0500
+++ llvm-2.5/lib/System/Unix/Signals.inc	2009-03-04 13:56:32.804820443 -0500
@@ -78,7 +78,7 @@
   for (int i = 0; i < depth; ++i) {
     Dl_info dlinfo;
     dladdr(StackTrace[i], &dlinfo);
-    char* name = strrchr(dlinfo.dli_fname, '/');
+    const char* name = strrchr(dlinfo.dli_fname, '/');
 
     int nwidth;
     if (name == NULL) nwidth = strlen(dlinfo.dli_fname);
@@ -93,7 +93,7 @@
 
     fprintf(stderr, "%-3d", i);
 
-    char* name = strrchr(dlinfo.dli_fname, '/');
+    const char* name = strrchr(dlinfo.dli_fname, '/');
     if (name == NULL) fprintf(stderr, " %-*s", width, dlinfo.dli_fname);
     else              fprintf(stderr, " %-*s", width, name+1);
 
