--- tools/llvmc/plugins/Base/Base.td.orig	2009-03-28 21:25:47.000000000 +0100
+++ tools/llvmc/plugins/Base/Base.td	2009-03-28 21:26:29.000000000 +0100
@@ -113,7 +113,7 @@
 [(in_language "llvm-bitcode"),
  (out_language "assembler"),
  (output_suffix "s"),
- (cmd_line "llc -relocation-model=pic -f $INFILE -o $OUTFILE"),
+ (cmd_line "llc -f $INFILE -o $OUTFILE"),
  (actions (case (switch_on "S"), (stop_compilation)))
 ]>;
 
