--- llvm-2.7/tools/clang/lib/Frontend/InitHeaderSearch.cpp.cxx_includes	2010-03-06 20:38:10.000000000 +0100
+++ llvm-2.7/tools/clang/lib/Frontend/InitHeaderSearch.cpp	2010-06-05 13:43:03.150438148 +0200
@@ -524,10 +524,32 @@
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.2",
                                 "i686-redhat-linux","", "", triple);
 
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.2",
+                                "x86_64-redhat-linux", "32", "", triple);
+
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.2",
+                                "ppc64-redhat-linux", "64", "", triple);
+
     // Fedora 12 (February-2010+)
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.3",
                                 "i686-redhat-linux","", "", triple);
-      
+
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.3",
+                                "x86_64-redhat-linux", "32", "", triple);
+
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.3",
+                                "ppc64-redhat-linux", "64", "", triple);
+
+    // Fedora 13
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.4",
+                                "i686-redhat-linux","", "", triple);
+
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.4",
+                                "x86_64-redhat-linux", "32", "", triple);
+
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.4.4",
+                                "ppc64-redhat-linux", "64", "", triple);
+
     // openSUSE 11.1 32 bit
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3",
                                 "i586-suse-linux", "", "", triple);
